src/sentry/releasehealth/metrics.py

```diff
@@ -347,6 +358,137 @@ def _count_users(total: bool, referrer: str) -> Dict[Any, int]:
 
         return rv
 
+    def get_release_sessions_time_bounds(
+        self,
+        project_id: ProjectId,
+        release: ReleaseName,
+        org_id: OrganizationId,
+        environments: Optional[Sequence[EnvironmentName]] = None,
+    ) -> ReleaseSessionsTimeBounds:
+        select: List[SelectableExpression] = [
+            Function("min", [Column("timestamp")], "min"),
+            Function("max", [Column("timestamp")], "max"),
+        ]
+
+        try:
+            where: List[Union[BooleanCondition, Condition]] = [
+                Condition(Column("org_id"), Op.EQ, org_id),
+                Condition(Column("project_id"), Op.EQ, project_id),
+                Condition(Column(tag_key(org_id, "release")), Op.EQ, tag_value(org_id, release)),
+                Condition(Column("timestamp"), Op.GTE, datetime.min),
+                Condition(Column("timestamp"), Op.LT, datetime.now(pytz.utc)),
+            ]
+
```

Comment:
Feels to me like the try/catch is wrapping much more logic than it actually should

I feel like lines 382 - 391 should be at the top of the function and if no `env_filter` exists just return `{"sessions_lower_bound": None, "sessions_upper_bound": None}` early on without raising an error and then catching it unless you want the error to be captured by sentry 
to which I would also suggest that if that is your goal with this to maybe use `logging.exception` or `metrics.incr`

Reply:
The point of the exception mgmt is that I can use `tag_key` and `tag_value` multiple times without having to check their return-value for `None` each. In previous PRs I would have to do that explicitly.

Instead of raising `MetricIndexNotFound()` on line 391 I can also return the same value, but I feel that it would be simply duplication of code.